### ISSUE TITLE
Warnings on build.

### ISSUE THREAD

**Issue by philderbeast** [bgamari/the-thoralf-plugin#1]
When I build, I see warnings.

```
the-thoralf-plugin> stack install
...
-Wunused-matches
-Wunticked-promoted-constructors
-Wdodgy-imports
-Wunused-imports
-Wunused-local-binds
...
thoralf-plugin-0.1.0.0: copy/register
Registering library for thoralf-plugin-0.1.0.0..
Completed 3 action(s).
```

**Comment by philderbeast** [comment]
I'd happily fix these warnings if you want to assign this issue to me.

**Comment by goldfirere** [comment]
It's yours. Thanks! :)

**Comment by bgamari** [comment]
I fix a number of warnings in #5.
